utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -0,0 +1,278 @@
+      case x: java.lang.Short => java.math.BigInteger.valueOf(x.intValue())
+      case x: Integer => java.math.BigInteger.valueOf(x.intValue())
+      case x: java.lang.Long => java.math.BigInteger.valueOf(x)
+      case x: java.math.BigInteger => x
+      case x => throw new IllegalArgumentException(s"Can't convert to BigInteger, unsupported number of type ${x.getClass.getName}: $x")
+    }
+  }
+
+  @Documentation(description = "Returns a sum of all elements")
+  @GenericType(typingFunction = classOf[ListElementTyping])
+  def sum[T <: java.lang.Number](@ParamName("listOfNumbers") list: java.util.Collection[T]): T = {
+    if (list.isEmpty) {
+      return 0.asInstanceOf[T]
+    }
+
+    val targetType = list.asScala.reduce { (a, b) =>
+      if (getNumberPriority(a) > getNumberPriority(b)) a else b
+    }
+
+    targetType match {
+      case _: java.lang.Byte => list.asScala.map(_.byteValue()).sum.asInstanceOf[T]
+      case _: java.lang.Short => list.asScala.map(_.shortValue()).sum.asInstanceOf[T]
+      case _: Integer => list.asScala.map(_.intValue()).sum.asInstanceOf[T]
+      case _: java.lang.Long => list.asScala.map(_.longValue()).sum.asInstanceOf[T]
+      case _: java.math.BigInteger => list.asScala.map(toBigInteger(_)).reduce((a, b) => a.add(b)).asInstanceOf[T]
+      case _: java.lang.Float => list.asScala.map(_.floatValue()).sum.asInstanceOf[T]
+      case _: java.lang.Double => list.asScala.map(_.doubleValue()).sum.asInstanceOf[T]
+      case _: java.math.BigDecimal => list.asScala.map(toBigDecimal(_)).reduce((a, b) => a.add(b)).asInstanceOf[T]
+      case x => throw new IllegalArgumentException(s"Can't calculate the sum of elements. Unexpected element of type ${x.getClass.getName}: $x")
+    }
+  }
+
+  @Documentation(description = "Returns a list of all elements sorted in ascending order (elements must be comparable)")
+  @GenericType(typingFunction = classOf[ListTyping])
+  def sortedAsc[T <: Comparable[T]](@ParamName("list") list: java.util.Collection[T]): java.util.List[T] = sorted(list, desc = false)
+
+  @Documentation(description = "Returns a list of all elements sorted in descending order (elements must be comparable)")
+  @GenericType(typingFunction = classOf[ListTyping])
+  def sortedDesc[T <: Comparable[T]](@ParamName("list") list: java.util.Collection[T]): java.util.List[T] = sorted(list, desc = true)
+
+  private def sorted[T <: Comparable[T]](list: java.util.Collection[T], desc: Boolean): java.util.List[T] = {
+    checkIfComparable(list)
+    val values = new java.util.ArrayList[T](list)
+    Collections.sort[T](values)
+    if (desc) {
+      Collections.reverse(values)
+    }
+    values
+  }
+
+  @GenericType(typingFunction = classOf[MapTyping])
+  @Documentation(description = "Returns a map with elements sorted by value in ascending order")
+  def sortedMapByValueAsc[K, V <: Comparable[V]](@ParamName("map") map: java.util.Map[K, V]): java.util.Map[K, V] = {
```

Comment:
Do we need this method? It will work, but our UI can't display ordered maps (our JSON serializer will output it in "random" order, so users can't really test this method)

Reply:
I;m totally convinced :) Removed